src/production/ProductionContext.ts

```diff
@@ -1,30 +1,51 @@
 import { Context } from '../core/Context';
+import { prerequisites } from '../main/prerequisite';
 import { IModule } from '../moduleResolver/Module';
+import { ModuleResolver } from '../moduleResolver/ModuleResolver';
+import { pluginAssumption } from '../plugins/core/plugin_assumption';
+import { pluginCSS } from '../plugins/core/plugin_css';
+import { pluginSass } from '../plugins/core/plugin_sass';
 import { ModuleTree } from './module/ModuleTree';
-import { SplitEntries, ISplitEntries } from './module/SplitEntries';
+import { createSplitEntries, ISplitEntries } from './module/SplitEntries';
+
+export interface IProductionContext {
+  ctx: Context;
+  entries?: Array<IModule>;
+  modules?: Array<IModule>;
+  splitEntries?: ISplitEntries;
+}
+
+export function createProductionContext(ctx: Context, modules?: Array<IModule>): IProductionContext {
+  let entries;
+
+  /**
+   * @todo:
+   * Not sure about this check
+   */
+  if (!modules) {
```

Comment:
from tests we provide it with a modules array...
do you have an idea how to solve this more properly?

Reply:
we need to throw an exception... Probably in ModuleResolver, a fatal error that the files weren't resolved